src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -2332,5 +2375,29 @@ private void AddMemberLookupSymbolsInfoInTypeParameter(LookupSymbolsInfo result,
                 AddMemberLookupSymbolsInfoWithoutInheritance(result, baseInterface, options, originalBinder, accessThroughType: type);
             }
         }
+
+        internal void AddImplicitExtensionMemberLookupSymbolsInfoForType(LookupSymbolsInfo result, TypeSymbol type, LookupOptions options, Binder originalBinder)
+        {
+            var accessThroughType = type;
+            if (type.ExtendedTypeNoUseSiteDiagnostics is { } extendedType)
```

Comment:
>if (type.ExtendedTypeNoUseSiteDiagnostics is { } extendedType) [](http://example.com/codeflow?start=12&length=62)

I am not sure why are we continuing otherwise. Could you elaborate, please.

Reply:
Perhaps even a better question: "Why would we ever call this method with an extension type?"

Reply:
Extension types get the benefit of members from other extension types too.
```
class C { }
implicit extension E for C { }
implicit extension E2 for C { void Member() { } }
```
When trying to list all the extension members that are applicable to type `E`, we should find `E2.Member`.
Note: we also do this during binding (in `LookupImplicitExtensionMembersInSingleBinder`).